src/app/garfile/departure/post.controller.js

```diff
@@ -49,16 +46,16 @@ const createValidationChains = (voyage) => {
     [new ValidationRule(validator.realDate, 'departureDate', departDateObj, realDateMsg)],
     [new ValidationRule(validator.currentOrFutureDate, 'departureDate', departDateObj, futureDateMsg)],
     [new ValidationRule(validator.validTime, 'departureTime', departureTimeObj, timeMsg)],
+    [new ValidationRule(validator.notEmpty, 'portChoice', voyage.portChoice, portChoiceMsg)],
   ];
 
   // Check if port code is ZZZZ then need to validate lat/long and display req zzzz message
   if (departPortObj.portCode.toUpperCase() === 'YYYY') {
     validations.push(
-      departurePortZZZZ,
       departureLatValidation,
       departureLongValidation,
     );
-  } else {
+  } else if (voyage.portChoice) {
```

Comment:
If this was not changed, then the validations would return both the port choice not being made and the departure port not being set, when only the first one makes sense to a user.